_posts/2017-07-06-hitchhikers-guide-to-cp.md

```diff
@@ -0,0 +1,308 @@
+---
+layout: post
+title:  "Hitchhiker's Guide to Competitive Programming"
+date:   2017-07-06
+author: udit
+---
+
+# Introduction
+
+Competitive programming, as the name suggests, is a sport of programming. One is presented with few problems for which one has to code solutions that should be efficient enough to fit in the time and memory constraints. Online judges like Codechef, Codeforces, Topcoder etc frequently organize programming competitions that one can take part in and win prizes. Few prestigious competitions like [ACM-ICPC](https://en.wikipedia.org/wiki/ACM_International_Collegiate_Programming_Contest), [Google Code Jam](https://en.wikipedia.org/wiki/Google_Code_Jam) and [Facebook Hacker Cup](https://en.wikipedia.org/wiki/Facebook_Hacker_Cup) are held annually. Doing great in these contests is an achievement in itself.
+
+Now the real question, **"Why should one try it?"**. Well, the answer is simple, programming is a major portion of the work that is done in tech companies. Being good at programming makes it easier to get a job in a good tech company. Competitive programming is the best way of getting good at programming and problem solving. Apart from this, one gets recognition and meets other programmers. There are prizes for top performers in each of these competitions. Most importantly, it quenches the thirst for those looking for challenges.
```

Comment:
I would emphasize the fact that it is fun and the closest you'll come to actual innovative computer science in the college here a bit more than the jobs stuff.